src/importer/DND5SpellImporter.ts

```diff
@@ -34,34 +34,68 @@ export class DND5SpellImporter {
 
     public importSpellsByUrl = async (url: string): Promise<void> => {
         try {
-            const response: multiSpellImport = await axios.get(url);
-            this.importSpellsByData(response.results);
+            axios.get(url).then(response => {
+                const spellLocations: multiSpellImport = response.data;
+                const spells: spellImport[] = [];
+                // collect errors that occur during the import
+                const errorList: string[] =[];
+                spellLocations.results.forEach(async location => {
+                    try {
+                        axios.get(`http://dnd5eapi.co${location.url}`).then(resp => {
+                        spells.push(resp.data);
+                        // check if last spell received
```

Comment:
sobald du beschreiben musst was etwas tut, pack es in eine private funktion und beschreibe es nicht mit einem Kommentar

Reply:
Das ist die if abfrage, damit man nicht lage überlegen muss. Ich kann gerne noch mehr private funktionen machen,halte das aber nach wie vor für overengineered. Kommentare sind genau für so etwas.

Reply:
Kommentare im code sind scheisse

Reply:
private funktionen an soner Stelle erleichtern die Lesbarkeit viel besser